packages/pwa-buildpack/src/WebpackTools/plugins/DevServerReadyNotifierPlugin.js

```diff
@@ -0,0 +1,47 @@
+const url = require('url');
+const boxen = require('boxen');
+const chalk = require('chalk');
+const once = require('one-time');
+const PostCompileWebpackPlugin = require('post-compile-webpack-plugin');
+
+class DevServerReadyNotifierPlugin {
+    constructor(devServer) {
+        if (!devServer || !devServer.host || !devServer.port) {
+            throw Error(
+                'Argument to DevServerReadyNotifierPlugin must be an object with host and port properties.'
+            );
+        }
+        this.url = url.format({
+            protocol: 'https',
+            // webpack-dev-server does not comply with the WHATWG URL
+            // format specification that `host` includes a port number,
+            // and `hostname` does not
+            hostname: devServer.host,
+            port: devServer.port
+        });
+    }
+    apply(compiler) {
+        const handler = once(() =>
+            setImmediate(() =>
+                console.log(
+                    boxen(
+                        chalk.green(
+                            'PWADevServer ready at ' +
+                                chalk.greenBright.underline(this.url)
+                        ),
+                        {
+                            borderColor: 'gray',
+                            float: 'center',
+                            align: 'center',
+                            margin: 1,
+                            padding: 1
+                        }
```

Comment:
Is this CSS for Terminal?

Reply:
It's the API signature of [boxen](https://www.npmjs.com/package/boxen), so I guess so, in that it's a box model.